sympy/solvers/ode.py

```diff
@@ -5788,6 +5789,19 @@ def ode_lie_group(eq, func, order, match):
             sol = solve(eq, df)
             if sol == []:
                 raise NotImplementedError
+            elif len(sol)>1:
+                sols = []
+                for i in sol:
+                    try:
+                        soln = ode_lie_group(Eq(df, i), func, order, match)
+                        sols.append(soln)
+                    except (ValueError, NotImplementedError):
+                        pass
+                if sols == []:
+                    raise NotImplementedError("Unable to solve the differential equation " +
+                str(eq) + " by the lie group method")
```

Comment:
I guess the message here isn't needed since this exception is just caught below. I would probably rearrange this logic though to something like:
```python
try:
    eqsol = solve(eq, df)
except NotImplementedError:
    eqsol = []

desols = []
for s in eqsol:
    try:
        # _ode_lie_group is a new function containing the lower part of this function
        desols.extend(_ode_lie_group(s, func, order, match))
    except (ValueError, NotImplementedError): # Why catch ValueError?
        pass

if desols = []:
    raise NotImplementedError("Unable...")

return desols
```
Do you know why ValueError is being caught? We probably shouldn't catch that...

Reply:
I would prefer an approach that doesn't involve raising and catching ValueError since other things can raise ValueError so this can end up hiding bugs.

Reply:
It's hard to inerpret such a large diff. Can you push the changes to the PR?